ovh/types_dedicated_server.go

```diff
@@ -111,24 +111,39 @@ type DedicatedServerInstallTaskCreateOpts struct {
 	TemplateName        string                             `json:"templateName"`
 	PartitionSchemeName *string                            `json:"partitionSchemeName,omitempty"`
 	Details             *DedicatedServerInstallTaskDetails `json:"details"`
+	UserMetadata        *[]map[string]string               `json:"userMetadata"`
 }
 
 func (opts *DedicatedServerInstallTaskCreateOpts) FromResource(d *schema.ResourceData) *DedicatedServerInstallTaskCreateOpts {
 	opts.TemplateName = d.Get("template_name").(string)
 	opts.PartitionSchemeName = helpers.GetNilStringPointerFromData(d, "partition_scheme_name")
 
 	details := d.Get("details").([]interface{})
-	if details != nil && len(details) == 1 {
+	if len(details) == 1 {
 		opts.Details = (&DedicatedServerInstallTaskDetails{}).FromResource(d, "details.0")
+	}
 
+	userMetadata := d.Get("user_metadata").([]interface{})
+	var userMetadatas []map[string]string
+	if len(userMetadata) >= 1 {
```

Comment:
why this if ?

Reply:
to check if userMetadata is not null ? otherwise we don't won't to set this field

Reply:
if `len(userMetadata) == 0`, you won't range in the `for` below, and `userMetadatas` will still be nil, so not sent to the API thanks to the `omitempty`, should be ok ?